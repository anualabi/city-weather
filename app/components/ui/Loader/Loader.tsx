import {
  ActivityIndicator,
  ActivityIndicatorProps,
  StyleSheet,
  View,
} from "react-native";

import { colors } from "@/theme";

export interface LoaderProps extends ActivityIndicatorProps {
  animating: boolean;
  color?: string;
  size?: number | "small" | "large";
}

export default function Loader({
  animating,
  color = colors.bgBlack,
  size = "large",
  ...props
}: LoaderProps) {
  return (
    <View style={styles.container} testID="loader">
      <ActivityIndicator
        animating={animating}
        color={color}
        size={size}
        testID="activity-indicator"
        {...props}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: { alignItems: "center", flex: 1, justifyContent: "center" },
});
