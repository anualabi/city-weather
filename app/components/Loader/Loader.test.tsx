import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import "@testing-library/jest-native/extend-expect";

import Loader, { LoaderProps } from "./Loader";

const customRender = (props: LoaderProps) => {
  return render(<Loader {...props} />);
};

describe("Loader component", () => {
  test("should render correctly", () => {
    const tree = renderer.create(<Loader animating={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render correctly", () => {
    const { getByTestId } = customRender({ animating: true });
    expect(getByTestId("loader")).toBeDefined();
  });

  test("should show ActivityIndicator when animating is true", () => {
    const { getByTestId } = customRender({ animating: true });
    expect(getByTestId("activity-indicator")).toBeDefined();
  });

  test("should hide ActivityIndicator when animating is false", () => {
    const { getByTestId } = customRender({ animating: false });
    const activityIndicator = getByTestId("activity-indicator");
    expect(activityIndicator.props.animating).toBe(false);
  });
});
