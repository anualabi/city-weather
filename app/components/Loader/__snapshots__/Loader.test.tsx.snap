// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="loader"
>
  <ActivityIndicator
    animating={false}
    color="black"
    size="large"
    testID="activity-indicator"
  />
</View>
`;
